### ISSUE TITLE
No activities found when using bounded_by

### ISSUE THREAD

**Issue by double-a** [NathanBaulch/rainbow-roads#7]
Trying to get it running, thus starting without bound_by (and crashing).

```
> ./bin/linux/rainbow-roads     -after 2022-01-01     -min_duration 15m     -min_distance 3km      -output test  ~/Downloads/Polar/gpx/*gpx
activity files: 774
 100% |████████████████████████████████████████| [25s:0s]             
activities:     39
period:         33.0 second(s) (2022-02-05 to 2022-02-05)
duration range: 40m37s to 2h11m41s
distance range: 3.6km to 51.7km
bounds:         48.32298,12.23296,39798.09228
total points:   151,978
total duration: 42h16m32s
total distance: 499.6km
panic: image: NewPaletted Rectangle has huge or negative dimensions

goroutine 1 [running]:
image.pixelBufferLength(0xc000279d28, {{0xc000279d40, 0x5a4c6d}, {0x669460, 0xc000080400}}, {0x613c1b, 0x0})
	/usr/local/go/src/image/image.go:80 +0xc5
image.NewPaletted({{0x0, 0xed988ae51}, {0x0, 0x0}}, {0xc0001fb000, 0x100, 0x100})
	/usr/local/go/src/image/image.go:1266 +0x73
main.render()
	/home/andreas/project/rainbow-roads-main/main.go:443 +0x2e5
main.main()
	/home/andreas/project/rainbow-roads-main/main.go:125 +0x435
```
Now trying to use bounded_by (similar to bounds in output, thus I assume latitude,longitude,radius which matches)

```
> ./bin/linux/rainbow-roads     -after 2022-01-01     -min_duration 15m     -min_distance 3km  -bounded_by 48.32,12.23,10km    -output test  ~/Downloads/Polar/gpx/*gpx
activity files: 774
 100% |████████████████████████████████████████| [25s:0s]             
2022/03/03 23:40:21 no matching activities found
```

Same result when using `-bounded_by 48.32,12.23,25000`

**Comment by NathanBaulch** [comment]
Thanks for the crash report. Firstly, what version are you running?

What happens if you specify `-bounded_by 48.32,12.23,40km`? The entire activity must fall within that circular region, so using a radius of 10km would likely exclude a bunch of activities. You might want to consider using the `passes_through` filter instead?

**Comment by double-a** [comment]
Downloaded the version from github yesterday ;-)
Unfortunately same result with
``` 
./bin/linux/rainbow-roads     -after 2022-01-01     -min_duration 15m     -min_distance 3km  -bounded_by 48.32,12.23,40km -sport running    -output test  ~/Downloads/Polar/gpx/*gpx
activity files: 779
 100% |████████████████████████████████████████| [25s:0s]             
2022/03/04 21:25:19 no matching activities found
```
and using `passes_through` (with same parameters) gives the same result 
```
./bin/linux/rainbow-roads     -after 2022-01-01     -min_duration 15m     -min_distance 3km  -passes_through 48.32,12.23,40km -sport running    -output test  ~/Downloads/Polar/gpx/*gpx
activity files: 779
 100% |████████████████████████████████████████| [25s:0s]             
2022/03/04 21:27:20 no matching activities found
```
Thanks for checking ...

**Comment by NathanBaulch** [comment]
I recommend removing all the filters and adding them back one at a time to see what's invalidating all your activities.

**Comment by double-a** [comment]
No filter crashes due to rectangle:
```
./bin/linux/rainbow-roads -output test  ~/Downloads/Polar/gpx/*gpx
activity files: 779
 100% |████████████████████████████████████████| [23s:0s]             
activities:     780
period:         3.6 year(s) (2018-07-21 to 2022-03-03)
duration range: 4s to 5h41m25s
distance range: 0.0km to 141.5km
bounds:         46.18611,10.90865,370083.00587
total points:   3,207,258
total duration: 905h45m17s
total distance: 14,698.2km
panic: image: NewPaletted Rectangle has huge or negative dimensions

goroutine 1 [running]:
image.pixelBufferLength(0xc0000d7d28, {{0xc0000d7d40, 0x5a4c6d}, {0x669460, 0xc000080000}}, {0x613c1b, 0x0})
	/usr/local/go/src/image/image.go:80 +0xc5
image.NewPaletted({{0x0, 0xed99f1ca5}, {0x0, 0x0}}, {0xc0120d4000, 0x100, 0x100})
	/usr/local/go/src/image/image.go:1266 +0x73
main.render()
	/home/andreas/project/rainbow-roads-main/main.go:443 +0x2e5
main.main()
	/home/andreas/project/rainbow-roads-main/main.go:125 +0x435
```

With the only filter `-sport running` nothing is found.
With the only filter `-after 2022-01-01` it finds 44 activities but crashes as with no filter. Same result when adding `-min_duration 15m` and adding `-min_distance 3km`.
Adding `-passes_through 48.32,12.23,40000` limits to 43 activities, still crashing with same backtrace.
Same result for `-bounded_by 48.32,12.23,40000`.

Thus I suspect that there is something with the Polar/gpx files which is not checked in the first pass but in the second.
Limiting to a single file shows exactly the same issue.

Attached a file for further analysis. BTW, same issue with tcx.
[polar-gpx.zip](https://github.com/NathanBaulch/rainbow-roads/files/8190753/polar-gpx.zip)

**Comment by NathanBaulch** [comment]
Thanks for the file, this is bug #2 which was fixed in v0.1.1 last week.
